tests/integration/test_charm.py

```diff
@@ -77,29 +77,66 @@ async def test_mesh_config(ops_test: OpsTest):
 
 
 @pytest.mark.abort_on_fail
-async def test_service_mesh_relation(ops_test: OpsTest, service_mesh_tester):
+async def test_deploy_service_mesh_apps(ops_test: OpsTest, service_mesh_tester):
     assert ops_test.model
-    # Ensure model is on mesh
-    await ops_test.model.applications[APP_NAME].set_config({"model-on-mesh": "true"})
-    time.sleep(5)  # Wait for the model to be on mesh
-
     # Deploy tester charms
     resources = {"echo-server-image": "jmalloc/echo-server:v0.3.7"}
     # Applications that will be given authorization policies
     # receiver1 require trust because the service-mesh library interacts with k8s objects.
     await ops_test.model.deploy(
-        service_mesh_tester, application_name="receiver1", resources=resources, trust=True
+        service_mesh_tester,
+        application_name="receiver1",
+        resources=resources,
+        trust=True,
     )
     await ops_test.model.deploy(
-        service_mesh_tester, application_name="sender1", resources=resources, trust=True
+        service_mesh_tester,
+        application_name="sender1",
+        resources=resources,
+        trust=True,
     )
     await ops_test.model.deploy(
-        service_mesh_tester, application_name="sender2", resources=resources, trust=True
+        service_mesh_tester,
+        application_name="sender2",
+        resources=resources,
+        trust=True,
     )
+
     await ops_test.model.add_relation("receiver1:service-mesh", APP_NAME)
+    await ops_test.model.add_relation("sender1:service-mesh", APP_NAME)
     await ops_test.model.add_relation("receiver1:inbound", "sender1:outbound")
+    await ops_test.model.wait_for_idle(
+        [
+            APP_NAME,
+            "receiver1",
+            "sender1",
+            "sender2",
+        ],
+        raise_on_error=False,
+    )
 
-    await ops_test.model.wait_for_idle([APP_NAME, "receiver1", "sender1", "sender2"])
+
+@pytest.mark.abort_on_fail
+@pytest.mark.parametrize("model_on_mesh", [True, False])
+async def test_service_mesh_relation(ops_test: OpsTest, model_on_mesh):
```

Comment:
I realize these didn't have docstrings before you were here, but lets start adding them so its clear for the next dev

Something like this?  Feel free to change

```suggestion
async def test_service_mesh_relation(ops_test: OpsTest, model_on_mesh):
    """Test that the service mesh relation correctly puts applications on the mesh and opens traffic as expected.
    
    The test sets `auto-join-mesh` for the tester charm based on the `model_on_mesh` parameter.  So:
    * when `model_on_mesh=True` we set `auto-join-mesh=False` to test that the model has subscribed the apps
    * when `model_on_mesh=False` we set `auto-join-mesh=True` to test that the apps have subscribed themselves
    """
```

Reply:
done! added docstrings to the test